### PR TITLE
Fixes being able to set shadowOptions to falsy values, adds shadowBlu…

### PR DESCRIPTION
**Pull request by Maistho**
* **Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

I didn't do any changes to tests or docs

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
When setting shadowOffsetX, if 0 is provided it instead uses 1 from the defaults.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
It fixes broken behaviour, if someone is relying on the broken behaviour then it will break for them.

* **Other information**:
I also added in the shadowBlur option, which was missing before

### CODE REVIEW

File: src/util/drawing.js [hunk 1]
```diff
@@ -90,14 +91,16 @@ export function path (context, options, fn) {
...
  */
 export function setShadow (context, options) {
   if (options.shadow) {
-    context.shadowColor = options.shadowColor || '#000000';
-    context.shadowOffsetX = options.shadowOffsetX || 1;
-    context.shadowOffsetY = options.shadowOffsetY || 1;
+    context.shadowColor = getDefault(options.shadowColor, '#000000');
+    context.shadowBlur = getDefault(options.shadowBlur, 0);
+    context.shadowOffsetX = getDefault(options.shadowOffsetX, 1);
+    context.shadowOffsetY = getDefault(options.shadowOffsetY, 1);
```
Comment (dannyrb) [inline src/util/drawing.js]: This makes more sense. Wanting to be able to have an offset of 0 wouldn't be possible without this method.

**Comment by dannyrb** [comment]
This looks like a safe and nice addition. I'll look more formally at merging it before Monday. `getDefault` makes sense.

File: src/util/drawing.js [hunk 2]
```diff
@@ -90,14 +91,16 @@ export function path (context, options, fn) {
...
+ * @param {Number} options.shadowBlur - Default Value: 0
  * @param {Number} options.shadowOffsetX - Default value: 1
  * @param {Number} options.shadowOffsetY - Default value: 1
  */
 export function setShadow (context, options) {
   if (options.shadow) {
-    context.shadowColor = options.shadowColor || '#000000';
-    context.shadowOffsetX = options.shadowOffsetX || 1;
-    context.shadowOffsetY = options.shadowOffsetY || 1;
+    context.shadowColor = getDefault(options.shadowColor, '#000000');
```
[the remark this answers is not in the document]
Comment (Maistho) [reply src/util/drawing.js]: Yes, with the string it doesn't make any difference, but I didn't want the code to be different for the strings vs numbers

**Comment by Maistho** [comment]
Do you want a similar pr for the vnext branch?

**Comment by dannyrb** [comment]
> Do you want a similar pr for the vnext branch?

No worries. I'll make the change.
Thanks for taking the time to create a PR! I'll try to be snappier about any future changes you propose ^_^

**Comment by Maistho** [comment]
Alright! Have a nice weekend =)

**Pull request merged**